ocaml/src/tools/prelude.ml

```diff
@@ -101,6 +101,16 @@ end
 module String = struct
   include String
 
+  let starts_with s prefix =
```

Comment:
This implementation is inefficient due to bounds checks, and it can throw `Invalid_argument 'index out of bounds'` when the prefix is longer than the string s.
Probably the easiest solution is to use `Core.String.is_prefix ~prefix s`